kardinal-manager/kardinal-manager/cluster_manager/cluster_manager.go

```diff
@@ -305,6 +306,13 @@ func (manager *ClusterManager) CleanUpClusterResources(ctx context.Context, clus
 		return nil
 	}
 
+	if isEmpty(clusterResources) {
```

Comment:
these changes come from this PR: https://github.com/kurtosis-tech/kardinal/pull/255